### PR TITLE
Fix sending emails to migrated subscribers

### PR DESCRIPTION
**Pull request by chao-xian**
The rake task for migrating subscribers to a different subscriber list
sent emails before the migration was completed. Especially for large
lists of users, this caused a confusing experience as emails were
received while the migration was still happening.

This change allows to to generate the emails to send, make the
migration, and then send the emails after the migration has
completed.

https://trello.com/c/HYqY4ZHw

### CODE REVIEW

File: app/services/bulk_email_sender_service.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class BulkEmailSenderService
-  def self.call(subject:, body:, subscriber_lists:, queue: :delivery_immediate)
-    email_ids = BulkEmailBuilder.call(subject: subject, body: body, subscriber_lists: subscriber_lists).ids
+  def self.call(bulk_email_builder:, queue: :delivery_immediate)
```
Comment (benthorner) [inline app/services/bulk_email_sender_service.rb:2]: We also need to change [this task](https://github.com/alphagov/email-alert-api/blob/master/lib/tasks/bulk.rake). I was tempted to remove it, but [it's an official thing](https://docs.publishing.service.gov.uk/manual/bulk-email.html). Maybe we should add a test for it, so it won't break in the future.

Comment (chao-xian) [reply app/services/bulk_email_sender_service.rb:2]: Ah good spot - thanks! Yes will add a test.

Comment (chao-xian) [reply app/services/bulk_email_sender_service.rb:2]: @benthorner [added a test for this rake task](https://github.com/alphagov/email-alert-api/pull/1261/commits/4c9951f5050a65629cb53b5138733389fe7fb19f)

File: spec/lib/tasks/bulk_spec.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+require "rails_helper"
+
+RSpec.describe "bulk" do
+  describe "email" do
+    before do
+      Rake::Task["bulk:email"].reenable
+      allow(ENV).to receive(:fetch).with("SUBJECT").and_return("subject")
```
Comment (benthorner) [inline spec/lib/tasks/bulk_spec.rb:7]: This repo [has ClimateControl](https://github.com/alphagov/email-alert-api/blob/master/Gemfile#L29), so we should use that here.

Comment (chao-xian) [reply spec/lib/tasks/bulk_spec.rb]: [Switched to ClimateControl](https://github.com/alphagov/email-alert-api/commit/2ee6d188f09d57f52f547629c8f5f9864a518752)

File: spec/lib/tasks/bulk_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+RSpec.describe "bulk" do
+  describe "email" do
+    before do
+      Rake::Task["bulk:email"].reenable
+    end
+
+    it "sends emails to a subscription list" do
+      subscriber_list = create(:subscriber_list)
+
+      ClimateControl.modify(SUBJECT: "subject", BODY: "body") do
```
Comment (benthorner) [inline spec/lib/tasks/bulk_spec.rb:12]: (You can also do this in an `around` block.)

Comment (chao-xian) [reply spec/lib/tasks/bulk_spec.rb]: [👍 ](https://github.com/alphagov/email-alert-api/commit/500fd67f01c9d5005dd49870459ba8c195ef0b05)

**Pull request merged**